### PR TITLE
feat: add web3profile to spa

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # mf-4489

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/app/src/components/DashboardDesktop.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ const navigation = [
     { name: 'Swap', icon: ArrowPathRoundedSquareIcon, to: ApplicationRoutes.Swap },
     { name: 'Bridges', icon: TruckIcon, to: ApplicationRoutes.Bridges },
     { name: 'Settings', icon: Cog6ToothIcon, to: ApplicationRoutes.Settings },
+    { name: 'web3profile', icon: ArrowPathRoundedSquareIcon, to: ApplicationRoutes.Web3Profile },
```
Comment (guanbinrui) [inline packages/app/src/components/DashboardDesktop.tsx]:
```suggestion
    { name: 'Web3 Profile', icon: ArrowPathRoundedSquareIcon, to: ApplicationRoutes.Web3Profile },
```

File: packages/app/src/constants/ApplicationRoutes.ts [hunk 2]
```diff
@@ -5,4 +5,5 @@ export enum ApplicationRoutes {
     Swap = '/swap',
     Bridges = '/bridges',
     Settings = '/settings',
+    Web3Profile = '/web3profile',
```
Comment (guanbinrui) [inline packages/app/src/constants/ApplicationRoutes.ts]:
```suggestion
    Web3Profile = '/web3-profile',
```

File: packages/shared/src/hooks/usePluginHostPermission.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react'
+import { useAsyncFn, useAsyncRetry } from 'react-use'
+import type { Plugin } from '@masknet/plugin-infra'
+import { MaskMessages } from '@masknet/shared-base'
+import { useSNSAdaptorContext } from '@masknet/plugin-infra/content-script'
+
+export function usePluginHostPermissionCheck(plugins: Plugin.Shared.Definition[]) {
```
Comment (Jack-Works) [inline packages/shared/src/hooks/usePluginHostPermission.ts:7]: why do you need to move it to the shared folder? app should not use it.

File: packages/mask/src/components/InjectedComponents/ProfileCard/index.tsx [hunk 4]
```diff
@@ -228,5 +228,3 @@ export const ProfileCard = memo(({ identity, currentAddress, ...rest }: Props) =
         </Web3ContextProvider>
     )
 })
-
-ProfileCard.displayName = 'ProfileCard'
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/ProfileCard/index.tsx]: don't remove this.

File: packages/shared/src/hooks/usePluginHostPermission.ts (hunk 3 above)
Comment (lelenei) [reply packages/shared/src/hooks/usePluginHostPermission.ts:7]: need

File: packages/app/src/mock.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import type { ProfileIdentifier } from '@masknet/base'
```
Comment (guanbinrui) [inline packages/app/src/mock.ts]:
```ts
/* cspell:disable */
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**